File: fenicia-web/src/types/basic/product/get-all-product-response.ts

export type GetAllProductResponse = {
    id: string;
    name: string;
    sku?: string;
    barcode?: string;
    description?: string;
    costPrice?: number;
    salesPrice: number;
    quantity: number;
    minStockLevel?: number;
    maxStockLevel?: number;
    imageUrl?: string;
    weight?: number;
    dimensions?: string;
    unitOfMeasure?: string;
    categoryId: string;
    categoryName: string;
    supplierId?: string;
    supplierName?: string;
    isActive: boolean;
};

File: Src/Libs/Fenicia.Common/DTOs/Basic/Product/GetAllProductResponse.cs
using System.ComponentModel.DataAnnotations;
using Fenicia.Common;

namespace Fenicia.Common.DTOs.Basic.Product;

public record GetAllProductResponse([Required] Guid Id,
    [Required] [MaxLength(200)] string Name,
    string? SKU,
    string? Barcode,
    string? Description,
    decimal? CostPrice,
    decimal SalesPrice,
    double Quantity,
    int? MinStockLevel,
    int? MaxStockLevel,
    string? ImageUrl,
    decimal? Weight,
    string? Dimensions,
    string? UnitOfMeasure,
    [Required] Guid CategoryId,
    [Required] [MaxLength(200)] string CategoryName,
    Guid? SupplierId,
    string? SupplierName,
    bool IsActive) : ICrudItem;